tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/internal/SourceLocationValidationService.java

```diff
@@ -45,6 +47,11 @@ protected Status compute() {
 			if ((sourceLocation == null) || sourceLocation.isEmpty() || !sourceLocation.isAbsolute()) {
 				retval = Status.createErrorStatus("Source code location must be a non-empty absolute path.");
 			}
+			else if (!SDKUtil.isValidSDKLocation(sourceLocation.toString()) &&
+					 !LiferayWorkspaceUtil.isValidWorkspaceLocation(sourceLocation.toString())) {
+
+				retval = Status.createErrorStatus("Source code location must be a SDK or LiferayWorkspace location.");
```

Comment:
We need to make this more dynamic.  We should probably use the ProjectImporter service.  We should make sure there is at least one ProjectImporter service that is able to import projects from this location.  You probably will need to add a method to the ProjectImporter interface something like `validate(Path)`